### ISSUE TITLE
Building selinux with AFLGo

### ISSUE THREAD

**Issue by arindam-8** [aflgo/aflgo#152]
Hi, 

I have been trying to build selinux with AFLGo for a research project. After running into a few issues, I have realised that the build error appears due to the usage of ```-fuse-ld=gold``` which I guess is important for AFLGo. I am able to build it using a regular compiler, but as soon as I add that flag, the build process fails complaining about missing symbols. I was wondering if you had any experience with this issue and/or if you could provide some assistance here. 

Thanks
Arindam

**Comment by mboehme** [comment]
Did you install [LLVM](https://releases.llvm.org/11.0.0/docs/CMake.html) 11.0.0 with [Gold](http://llvm.org/docs/GoldPlugin.html)-plugin and execute the following commands?

```sh
# Install LLVMgold into bfd-plugins
mkdir /usr/lib/bfd-plugins
cp /usr/local/lib/libLTO.so /usr/lib/bfd-plugins
cp /usr/local/lib/LLVMgold.so /usr/lib/bfd-plugins
```

**Comment by arindam-8** [comment]
Hi Marcel, 

Thanks for the response, really appreciate it! Yes indeed, I followed the setup instructions (inside the same container) and to verify the setup, I ran it on another benchmark (libjpeg-turbo) which worked just fine. 

I am investigating it on my side as well, but just wanted to check if this was a known issue with selinux before diving in. 

Thanks
Arindam

**Comment by mboehme** [comment]
Can you paste the build error here?

**Comment by arindam-8** [comment]
Hi Marcel, 

Here's the error:
```
make[1]: *** No rule to make target 'selinux-f7cb5901f1fb8be8fd038d64e54b5627bca5b88b/usr/lib/libsepol.a', needed by 'checkpolicy'.  Stop.
make[1]: Leaving directory '/benchmark/RUNDIR-secilc-f7cb5901f1fb8be8fd038d64e54b5627bca5b88b/selinux-f7cb5901f1fb8be8fd038d64e54b5627bca5b88b/checkpolicy'
make: *** [Makefile:34: install] Error 1
```
I suspect it is the manner in which the flags are being altered in **my** build setup (?) as I found [this](https://selinux.tycho.nsa.narkive.com/uOMinU6e/rework-of-makefiles#post44) post discussing certain implications within selinux. 

Just a clarification, this error does not appear while following the regular build instructions from the selinux [repository](https://github.com/SELinuxProject/selinux). 

Thanks
Arindam

**Comment by mboehme** [comment]
Yes. Given that make complains, this looks like an issue of overriding configuration.

**Comment by arindam-8** [comment]
Here is an update on the issue and how to at least partly work around it to build `SElinux` with `AFLGO`. It mainly stems from the pre-requisite regarding `libsepol`.

### Comments

- Instrumenting **libsepol.so** with AFLGo injects references to the fuzzer runtime (`__afl_area_ptr`, `__afl_prev_loc`, etc.), so linking both the shared library **and** the final executable with that runtime causes **multiple‑definition** errors.  
- Building **libsepol** as a **static** archive avoids duplicate symbols but produces a fully statically linked binary but requires fiddling with the build process of `SElinux`.  
- Relying on `-Wl,--allow-multiple-definition` silences linker complaints but is a _hack_.  

(Closing it with the comment)
